### PR TITLE
Dev w4m new tools

### PR DESCRIPTION
**Pull request by npaulhe**
add 4 new tools from w4m
- transformation
- batch correction
- generic filter
- quality metrics

### CODE REVIEW

**Comment by pcm32** [comment]
Hi guys! Great work! Given the closeness to the release, I'm compelled to ask what is the status of the containers bound to these tools? If those tool's containers don't have a release by Monday, then I wouldn't merge this until we branch to the release branch, and this is left as feature for the following release.

File: config/phenomenal_tools2container.yaml [hunk 1]
```diff
@@ -77,6 +77,39 @@ assignment:
...
+
+- docker_image_override: tool-generic_filter
+  docker_owner_override: phnmnl
+  docker_repo_override: container-registry.phenomenal-h2020.eu
+  docker_tag_override: vphenomenal_2017.12.12_cv0.2.1
+  max_pod_retrials: 3
+  tools_id:
+  - generic_filter
+
+- docker_image_override: qualitymetrics
```
Comment (pcm32) [inline config/phenomenal_tools2container.yaml]: Ok! Excellent, so only this container doesn't have a release, I think you are in good track guys!

Comment (pcm32) [reply config/phenomenal_tools2container.yaml]: This answers my above question.

**Comment by pcm32** [comment]
Excellent work @npaulhe! My only concern currently is that your github repos container-<tool-x> don't have all coherent naming at the Jenkins container build job (so tool-x should be written there in the same way), the Jenkins test job (test-container-<tool-x>) an container (plain <tool-x>). Since you have already build and pushed some containers, I suggest that you modify your github repos names and Jenkins jobs names to fit the pushed container image name for each tool. If you proceed like that, then it shouldn't impact this PR (which is as I see it ready to go, besides that thingy).

**Comment by pkrog** [comment]
I've only seen an issue with batch correction name.
I've checked names inside Jenkins and compared them with GitHub, but everything seems fine to me, all names are coherent.

**Comment by pkrog** [comment]
A tool is lacking: normalization. It is needed to complete the sacurine workflow. I will add it.

**Comment by pkrog** [comment]
Yes, it works, I've tested each tool.

**Pull request merged**